src/parser.cpp

```diff
@@ -50,7 +50,7 @@ namespace ada::parser {
     if (pointer_start == internal_input.end()) { pointer_start = internal_input.begin(); }
     std::string_view::iterator pointer_end = std::find_if_not(internal_input.rbegin(), std::make_reverse_iterator(pointer_start), ada::unicode::is_c0_control_or_space).base();
 
-    std::string_view url_data(pointer_start, pointer_end - pointer_start);
+    std::string_view url_data(&*pointer_start, pointer_end - pointer_start);
```

Comment:
Is there a better fix for this?

Reply:
Yes. We should have just one string_view throughout instead of working with pointers. But it requires much more engineering.

Reply:
We have all this trouble because the code is pointer-oriented. We don't need pointers at all!!!